engine/src/main/java/org/destinationsol/world/WorldBuilder.java

```diff
@@ -0,0 +1,104 @@
+
+
+    /**
+     * Initialize the WorldBuilder class
+     * @param context System Context
+     * @param numSystems Number of SolSystems the world should have. This can be set by the user
+     */
+    public WorldBuilder(Context context, int numSystems) {
+        this.context = context;
+        numberOfSystems = numSystems;
+        populateSolSystemGeneratorList();
+        populateFeatureGeneratorList();
+        initializeSolSystemGenerators();
+    }
+
+    /**
+     * This method uses reflection to retrieve all SolSystemGenerator classes. They are added to the list
+     * of SolSystemGenerators.
+     */
+    private void populateSolSystemGeneratorList() {
+
+        for (Class generator : context.get(ModuleManager.class).getEnvironment().getSubtypesOf(SolSystemGenerator.class)) {
+            try {
+                for (int i = 0; i < numberOfSystems; i++) {
+                    //for each SolSystem available, we will make the number of instances equal to the total number
+                    //of SolSystems we want for our world
+                    SolSystemGenerator solSystemGenerator = (SolSystemGenerator) generator.newInstance();
+                    solSystemGenerators.add(solSystemGenerator);
+                }
+            } catch (InstantiationException | IllegalAccessException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * This method uses reflection to retrieve all concrete FeatureGenerator classes. They are added to the list
+     * of FeatureGenerators.
+     */
+    private void populateFeatureGeneratorList() {
+        for (Class generator : context.get(ModuleManager.class).getEnvironment().getSubtypesOf(FeatureGenerator.class)) {
+            if (!Modifier.isAbstract(generator.getModifiers())) {
+                try {
+                    FeatureGenerator featureGenerator = (FeatureGenerator) generator.newInstance();
+                    featureGenerators.add(featureGenerator);
+                } catch (InstantiationException | IllegalAccessException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    /**
+     * This method initializes the SolSystemGenerators. How many generators are initialized depends on the number
+     * of SolSystems the world is set to have. When there are multiple types of SolSystemGenerators available, this
+     * method chooses randomly from the list of all generators to decide which to create.
+     */
+    private void initializeSolSystemGenerators() {
+        Random random = new Random();
+        for (int i = 0; i < numberOfSystems; i++) {
+            int systemIndex = random.nextInt(solSystemGenerators.size());
+            SolSystemGenerator solGenerator = solSystemGenerators.get(systemIndex);
+            solSystemGenerators.remove(systemIndex);
+            solGenerator.setFeatureGenerators(featureGenerators);
+            solGenerator.build();
```

Comment:
I think building the actual systems should be separate from the setup code. Even if it's called right after in `SolGame`, it still is conceptually separate, and moving it to a separate method call (`worldBuilder.build()`) allows for it to be created whenever makes the most sense.